packages/trpc/server/routers/viewer/holidays/checkConflicts.handler.ts

```diff
@@ -0,0 +1,119 @@
+import dayjs from "@calcom/dayjs";
+import { getHolidayService } from "@calcom/lib/holidays";
+import { CONFLICT_CHECK_MONTHS } from "@calcom/lib/holidays/constants";
+import prisma from "@calcom/prisma";
+import { BookingStatus } from "@calcom/prisma/enums";
+import type { TrpcSessionUser } from "@calcom/trpc/server/types";
+
+import type { TCheckConflictsSchema } from "./checkConflicts.schema";
+
+type CheckConflictsOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TCheckConflictsSchema;
+};
+
+export interface ConflictingBooking {
+  id: number;
+  title: string;
+  startTime: Date;
+  endTime: Date;
+  attendeeName: string | null;
+}
+
+export interface HolidayConflict {
+  holidayId: string;
+  holidayName: string;
+  date: string;
+  bookings: ConflictingBooking[];
+}
+
+export async function checkConflictsHandler({ ctx, input }: CheckConflictsOptions) {
+  const userId = ctx.user.id;
+  const { countryCode, disabledIds } = input;
+
+  if (!countryCode) {
+    return { conflicts: [] };
+  }
+
+  const startDate = new Date();
```

Comment:
let's avoid having logics and prisma calls directly within the handler. instead, move them into services & repositories.